File: Java/프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
import java.util.ArrayList;
class Solution {
    public int solution(String[] babbling) {
        ArrayList<String> po = new ArrayList<>();
        int result = 0;
        for(int i = 0; i<babbling.length;i++){
            if(!(babbling[i].contains("ayaaya") || babbling[i].contains("yeye") 
               || babbling[i].contains("woowoo") || babbling[i].contains("mama"))){
                po.add(babbling[i]);
            }
        }
        for(int i=0; i<po.size();i++){
            String a = po.get(i);
            if(!a.equals("")){
                a= a.replace("aya",":").replace("ye",":").replace("woo",":").replace("ma",":");
                a=a.replace(":","");
                result += a.equals("") ? 1 : 0 ;
            }    
        }
        
        return result;
    }
}

File: Kotlin/프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.kt
class Solution {
    val kotoba = arrayOf("aya", "ye", "woo", "ma")
    fun solution(babbling: Array<String>): Int = babbling.count{removeWord(it).isEmpty()}

    fun removeWord(word : String) : String{
        var result = word
        for(i in 0..kotoba.lastIndex) {
            if(word.startsWith(kotoba[i])) {
                result = word.replaceFirst(kotoba[i],"") 
                if(result.startsWith(kotoba[i])) {
                    return result
                }
            } 
        }
        return if(result==word) result else removeWord(result)
    }
}